cpp/CMakeLists.txt

```diff
@@ -40,27 +32,42 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif()
 
+if(${CMAKE_TOOLCHAIN_FILE} MATCHES "vcpkg.cmake$")
+    message(STATUS "CMAKE_TOOLCHAIN_FILE is set to vcpkg, so will use find_package to find the libraries")
+    set(ARCTICDB_USING_VCPKG ON)
+endif()
+
+if(NOT DEFINED ENV{CMAKE_BUILD_PARALLEL_LEVEL})
+    include(ProcessorCount)
+    ProcessorCount(_proc_count)
+    if(${_proc_count} LESS 2)
+        message(WARNING "Failed to determine the available processor count. Assuming you have at least 2 cores. \
+            Please set the CMAKE_BUILD_PARALLEL_LEVEL environment variable.")
+        set(ENV{CMAKE_BUILD_PARALLEL_LEVEL 2)
+    else()
+        cmake_host_system_information(RESULT _physical_mb QUERY TOTAL_PHYSICAL_MEMORY)
+        math(EXPR _needed_mb "${_proc_count} * 1500")
+        if(${_needed_mb} GREATER ${_physical_mb})
+            message(WARNING "We recommend 1500MB of physical RAM per processor core. We found ${_proc_count} cores and\
+                ${_physical_mb}MB of RAM. Use the CMAKE_BUILD_PARALLEL_LEVEL environment variable to avoid thrashing.")
+        endif()
+        set(ENV{CMAKE_BUILD_PARALLEL_LEVEL ${_proc_count})
+    endif()
+endif()
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-if(NOT BUILD_PYTHON_VERSION AND NOT WIN32)
-    message("Resetting Python include locations...")
-    # /default-medusa-venv is symlinked to /default-pegasus-venv
-    set(ENV_BASE_DIR "/default-medusa-venv")
-
-    set(BUILD_PYTHON_VERSION 3.6)
-
-    set(PYTHON_LIBRARIES "${ENV_BASE_DIR}/lib")
-    set(PYTHON_EXECUTABLE "${ENV_BASE_DIR}/bin/python")
-    set(PYTHON_INCLUDE_DIRS "${ENV_BASE_DIR}/include/python${BUILD_PYTHON_VERSION}m")
-    set(PYTHON_LIBRARY_SO "${ENV_BASE_DIR}/lib/libpython${BUILD_PYTHON_VERSION}m.so")
-endif()
+include(PythonUtils) # Must be called before Pybind (third_party) to override its finding mechanism
```

Comment:
I'm a bit lost here - how does this override the finding mechanism?

Reply:
By setting variables before pybind11 is included.